koswat/dike_reinforcements/reinforcement_profile/standard/standard_reinforcement_profile_builder.py

```diff
@@ -46,11 +50,13 @@ class StandardReinforcementProfileBuilder(ReinforcementProfileBuilderBase):
     def get_input_profile_calculator(
         reinforcement_type: type[StandardReinforcementProfile],
     ) -> ReinforcementInputProfileCalculationProtocol:
+        if issubclass(reinforcement_type, SoilReinforcementProfile):
+            return SoilInputProfileCalculation()
+        if issubclass(reinforcement_type, VPSReinforcementProfile):
```

Comment:
I hope this is being tested, I believe from other projects the following is also true:
`issubclass(reinforcement_type, VPSReinforcementProfile) and issubclass(reinforcement_type, SoilReinforcementProfile)`

If that's the case then you should first check whether `issubclass(reinforcement_type, SoilReinforcementProfile)` is `true` and within that `elif` statement have another conditional where you just check the specializations and otherwise return the parent one.

Which basically would imply you need to invert the order. Also, not sure again why you do not like `elif`.

Reply:
In general I think we agree to reduce complexity, which IMO is done by replacing `elif` by `if`, assuming each branch has a `return`. The execution with `elif` is done from top to bottom, so the effect is the same.
I think you are right that for VPS it will return a `SoilInputProfileCalculation` object, which should not be the case.
Adjusted the order.